ide/api.java.classpath/src/org/netbeans/api/java/classpath/ClassPath.java

```diff
@@ -241,7 +241,7 @@ public ClassPathImplementation getClassPathImpl(ClassPath cp) {
         long current;
         synchronized (this) {
             if (rootsCache != null) {
-                return this.rootsCache;
+                return Arrays.copyOf(rootsCache, rootsCache.length);
```

Comment:
I'd use `this.rootsCache.clone()`.

Reply:
Good point; recommitted as 3fc65de. Thanks.